File: Demos/Datasnap/Server/proxy/csharp_silverlight/TJSONString.cs
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

using System;

namespace Embarcadero.Datasnap.WindowsPhone7
{
    public class TJSONString : TJSONValue {
	    protected String value;

        /**
    	 * Class constructor, initialized the internal value to null
    	 */
	    public TJSONString() : base()
        {
		    value = null;
	    }

       /**
    	 * Class constructor, initialized the internal value with the value passed
    	 */
	    public TJSONString(String value) : base()
        {
		    this.value = value;
	    }

	    public override Object getInternalObject() {
		    return value;
	    }

	    public override String ToString() {
		    if (value != null)
			    return "\"" + value + "\"";
		    return NullString;
	    }

	    public override JSONValueType getJsonValueType() {
		    return JSONValueType.JSONString;
	    }
	
		/**
    	 * Gets the internal value 
    	 * @return String
    	 */
	    public String getValue(){
		    return value;
	    }
    }
}

File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TJSONString.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TJSONString.h"


@implementation TJSONString
-(id) init {
	self = [super init];
	if (self) {
		value =nil;
	}
	return self;
}
-(void)dealloc {
	[value release];
	[super dealloc];
}
-(id) initWithString:(NSString *) aValue{
	self = [self init];
	if (self) {
		[value release];
		value = [aValue retain];
	}
	return self;
}
-(id) getInternalObject{
	return value;
}

-(NSString *) toString{
  	if (value)
		return [NSString stringWithFormat:@"\"%@\"",value];
	return [self nullString];

}

-(JSONValueType) getJSONValueType{
	return JSONString;
}

-(NSString *) getValue {
	return value;
}

@end
@implementation TJSONString(jsonstringCreation)
+(id) JSONStringWithString:(NSString *) aValue{
	return [[[TJSONString alloc]initWithString:aValue]autorelease];
}

@end